lib/modules/apostrophe-assets/public/js/modules/demo.js

```diff
@@ -0,0 +1,99 @@
+$(function () {
+  var $body = $('body');
+  if ($body.hasClass('is-demo-mode--true is-logged-in')) {
+    var modalCookieName = 'apostrophe-demo-modal-3894eee4eee427';
```

Comment:
You don't have to go this road of obscure naming, the default domain for a cookie is hostname-specific, even to a subdomain:

https://stackoverflow.com/questions/18492576/share-cookie-between-subdomain-and-domain